services/gitdiff/gitdiff.go

```diff
@@ -536,6 +529,11 @@ parsingLoop:
 			lastFile := createDiffFile(diff, line)
 			diff.End = lastFile.Name
 			diff.IsIncomplete = true
+
+			// signal that we are exiting this diff early
```

Comment:
Maybe we should reuse `ctx` to do that?

Reply:
Can you elaborate on what that might look like? This is calling the `CancelFunc` of the provided `ctx`.

Reply:
This might be fixed in https://github.com/go-gitea/gitea/pull/32393/commits/e09aaadd170838419e093f42bbc6861d68d178f0. See other discussion.